File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        most = 0
        left = 0
        right = left + 1
        while right != len(prices):
            if prices[right] - prices[left] > most:
                most = prices[right] - prices[left]
            if right == len(prices) - 1:
                left += 1
                right = left + 1
            else:
                right += 1
        return most

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int max = Integer.MIN_VALUE;
        int left = 0, right = arr.length - 1;
        while (left < arr.length - 1) {
            if (arr[right] > max) max = arr[right];
            if (right == left + 1) {
                arr[left] = max;
                left++;
                right = arr.length - 1;
                max = Integer.MIN_VALUE;
            } else {
                right -= 1;
            }
        }
        arr[arr.length - 1] = -1;
        return arr;
    }
}